### PR TITLE
Update transfer learning docs

### PR DESCRIPTION
**Pull request by rohitgr7**
## What does this PR do?

Part of docs revamp #7666

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda

### CODE REVIEW

File: docs/source/advanced/transfer_learning.rst [hunk 1]
```diff
@@ -119,3 +128,52 @@ Here's a model that uses `Huggingface transformers <https://github.com/huggingfa
             h_cls = h[:, 0]
             logits = self.W(h_cls)
             return logits, attn
+
+
```
Comment (rohitgr7) [inline docs/source/advanced/transfer_learning.rst:132]:
```suggestion
----------

```

File: docs/source/advanced/transfer_learning.rst [hunk 2]
```diff
@@ -2,19 +2,26 @@
...
+################################
+Transfer Learning and Finetuning
+################################
 
-Using Pretrained Models
-^^^^^^^^^^^^^^^^^^^^^^^
+Transfer learning is a training methodology where an existing pre-trained model developed for another task is used as a backbone or a starting point
+to train models for new tasks. It is expected that since the existing pre-trained model is already a SOTA model, it can be used to generate better
+representations of the new data, given that the pre-trained model used was trained under similar domain. This technique can help generate better models
+with small dataset and models can generalize better or lead better results in less epochs as compared to training one from scratch.
```
Comment (tchaton) [inline docs/source/advanced/transfer_learning.rst:12]: I believe we could provide a better definition here and provide examples too.

Comment (tchaton) [reply docs/source/advanced/transfer_learning.rst:12]: Plus, the terminology should be introduced along side the definition.

File: docs/source/advanced/transfer_learning.rst [hunk 3]
```diff
@@ -119,3 +128,53 @@ Here's a model that uses `Huggingface transformers <https://github.com/huggingfa
...
             logits = self.W(h_cls)
             return logits, attn
+
+----------
+
+**********
+Finetuning
+**********
+
+For finetuning, Lightning provides a :class:`~pytorch_lightning.callbacks.finetuning.BackboneFinetuning` callback that can integrate the finetune strategy
```
Comment (tchaton) [inline docs/source/advanced/transfer_learning.rst:138]: I don't think we should mention `BackboneFinetuning`. This isn't really based on a strong heuristic and it is quite opinionated. I added this at a time when users need an example but I know personally feel we should depreciate it.

File: docs/source/advanced/transfer_learning.rst [hunk 4]
```diff
@@ -2,19 +2,26 @@
...
-^^^^^^^^^^^^^^^^^^^^^^^
+Transfer learning is a training methodology where an existing pre-trained model developed for another task is used as a backbone or a starting point
+to train models for new tasks. It is expected that since the existing pre-trained model is already a SOTA model, it can be used to generate better
+representations of the new data, given that the pre-trained model used was trained under similar domain. This technique can help generate better models
+with small dataset and models can generalize better or lead better results in less epochs as compared to training one from scratch.
 
-Sometimes we want to use a LightningModule as a pretrained model. This is fine because
-a LightningModule is just a `torch.nn.Module`!
+Finetuning is a technique usually done after transfer learning where after pre-training, the backbone model is unfronzen and trained for a few more epochs
+to improve the metric and make it more compatible with the data and current tasks, but this is not always true as sometimes it can lead to worse convergence.
```
Comment (tchaton) [inline docs/source/advanced/transfer_learning.rst:15]:
```suggestion
to improve the metric and make it more compatible with the data and current tasks, but this is not always true as sometimes it can lead to worse convergence. It is possible to have a more advanced scheduling fine-tuning strategy which can lead to better performance.  
```

File: docs/source/advanced/transfer_learning.rst [hunk 5]
```diff
@@ -2,19 +2,26 @@
...
 
-.. note:: Remember that a LightningModule is EXACTLY a torch.nn.Module but with more capabilities.
+*****************
+Transfer Learning
+*****************
 
-Let's use the `AutoEncoder` as a feature extractor in a separate model.
+Sometimes we want to use a LightningModule as a pretrained model. This is fine because
```
Comment (tchaton) [inline docs/source/advanced/transfer_learning.rst:21]: Let's link the LightningModule.

File: docs/source/advanced/transfer_learning.rst [hunk 6]
```diff
@@ -90,15 +97,17 @@ And use it to predict your data of interest
     model.freeze()
 
     x = some_images_from_cifar10()
+    x = images_to_tensor()
```
Comment (tchaton) [inline docs/source/advanced/transfer_learning.rst:100]: Wierd code. Let's use CifarDataset and transform. This code is confusing.

File: docs/source/advanced/transfer_learning.rst (hunk 3 above)
Comment (tchaton) [reply docs/source/advanced/transfer_learning.rst:138]: @rohitgr7 I would remove this one.

**Review by tchaton — APPROVED** [review]
LGTM !
